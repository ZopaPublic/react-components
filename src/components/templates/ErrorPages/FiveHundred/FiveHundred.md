### Summary

`<ErrorPageTemplate.FiveHundred />` renders the default content for when an error occurs

### Example

```tsx
import { ErrorPageTemplate } from '@zopauk/react-components';

<ErrorPageTemplate>
  <ErrorPageTemplate.FiveHundred />
</ErrorPageTemplate>;
```
