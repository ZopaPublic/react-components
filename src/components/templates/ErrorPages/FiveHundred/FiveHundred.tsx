import React from 'react';
import { faTools } from '@fortawesome/free-solid-svg-icons';

import FlexRow from '../../../layout/FlexRow/FlexRow';
import FlexCol from '../../../layout/FlexCol/FlexCol';
import Heading from '../../../atoms/Heading/Heading';
import Text from '../../../atoms/Text/Text';
import Link from '../../../atoms/Link/Link';
import { useViewport } from '../../../../hooks/useViewport/useViewport';
import { breakpoints } from '../../../../constants/breakpoints';
import { colors } from '../../../../constants';
import { ErrorTemplateProps } from '../types';
import { StyledIcon, StyledLink } from '../styles';
import { ThemeProvider, useThemeContext } from '../../../styles/Theme';

export interface FiveHundredErrorProps extends ErrorTemplateProps {
  linkUrl?: string;
  linkText?: string;
}

const FiveHundred = ({
  icon = faTools,
  linkUrl = 'https://www.zopa.com',
  linkText = 'Go to Zopa home',
}: FiveHundredErrorProps) => {
  const { width = 0 } = useViewport();
  const theme = useThemeContext();

  return (
    <div data-automation="ZA.ErrorPage.FiveHundred">
      <ThemeProvider theme={theme}>
        <FlexRow justify="center">
          <FlexCol xs="auto" className="mb-7">
            <StyledIcon bgColor={colors.greyLighter} variant={icon} size="3x" color={colors.greyDark} />
          </FlexCol>
        </FlexRow>
        <Heading as="h1" size={width <= breakpoints.desktop ? 'h4' : 'h3'} align="center" className="mb-4">
          Sorry, there’s been a problem
        </Heading>
        <Text as="p" size={width <= breakpoints.desktop ? 'body' : 'lead'} align="center" className="mb-7">
          We’re working to fix this problem and we’ll be up and running shortly.
        </Text>
        <Text as="p" size="small" align="center" className="mb-8">
          Keep seeing this page?{' '}
          <Link href="http://www.wikihow.com/Clear-Your-Browser%27s-Cookies" target="_blank" showTargetIcon={false}>
            Try clearing your browser cookies
          </Link>{' '}
          or come back later
        </Text>
        <FlexRow justify="center">
          <FlexCol xs="auto">
            <StyledLink className="mb-10" href={linkUrl} styling="secondary">
              {linkText}
            </StyledLink>
          </FlexCol>
        </FlexRow>
        <Text as="p" size="small" align="center">
          Error code: 500
        </Text>
      </ThemeProvider>
    </div>
  );
};

export default FiveHundred;
