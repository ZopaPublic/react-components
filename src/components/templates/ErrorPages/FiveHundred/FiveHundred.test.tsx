import React from 'react';
import { render } from '@testing-library/react';

import FiveHundred from './FiveHundred';

describe('<FiveHundred />', () => {
  it('renders with all the props', () => {
    const { container } = render(<FiveHundred />);
    expect(container).toMatchSnapshot();
  });
});
