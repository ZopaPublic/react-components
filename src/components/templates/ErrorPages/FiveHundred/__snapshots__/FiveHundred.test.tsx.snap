// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FiveHundred /> renders with all the props 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  margin-right: -12px;
  margin-left: -12px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c1 {
  position: relative;
  width: 100%;
  min-height: 1px;
  padding-right: 12px;
  padding-left: 12px;
  -webkit-align-self: auto;
  -ms-flex-item-align: auto;
  align-self: auto;
}

.c2 {
  background-color: #EFEFEF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 50%;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  padding: 24px;
}

.c4 {
  font-size: 24px;
  line-height: 32px;
  -webkit-letter-spacing: -0.25px;
  -moz-letter-spacing: -0.25px;
  -ms-letter-spacing: -0.25px;
  letter-spacing: -0.25px;
  margin: 0;
  color: #2C3236;
  text-align: center;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 800;
}

.c5 {
  margin: 0;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #2C3236;
  font-size: 16px;
  line-height: 24px;
  font-weight: 400;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  text-align: center;
}

.c6 {
  margin: 0;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  color: #2C3236;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  text-align: center;
}

.c7 {
  background-color: transparent;
  font-size: inherit;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  line-height: inherit;
  color: #3B46C4;
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-user-select: auto;
  -moz-user-select: auto;
  -ms-user-select: auto;
  user-select: auto;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding: 0;
  border: none;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  color: auto;
}

.c7:hover,
.c7:active {
  color: #21247F;
}

.c7:hover svg path,
.c7:active svg path {
  fill: #21247F;
}

.c7:hover {
  color: auto;
}

.c7:hover svg path {
  fill: auto;
}

.c7:active {
  color: auto;
}

.c7:active svg path {
  fill: auto;
}

.c3 {
  width: 80px;
  height: 80px;
}

.c8 {
  outline: 0;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  box-sizing: border-box;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 12px 24px;
  background: #E9EAFA;
  border-radius: 8px;
  border: 1px solid transparent;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  color: #21247F;
  font-size: 16px;
  font-weight: 600;
  line-height: 24px;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
}

.c8:focus:not(:active) {
  border: 1px solid #FFFFFF;
  box-shadow: 0 0 4px #3B46C4;
}

.c8:active {
  opacity: 0.8;
  box-shadow: unset;
  border: 1px solid transparent;
}

.c8:hover {
  background: #EEEFFB;
}

@media (min-width:0px) {
  .c1 {
    display: block;
    -webkit-flex: 0 0 auto;
    -ms-flex: 0 0 auto;
    flex: 0 0 auto;
    width: auto;
    max-width: none;
  }
}

<div>
  <div
    data-automation="ZA.ErrorPage.FiveHundred"
  >
    <div
      class="c0"
      cols="12"
      direction="row"
    >
      <div
        class="c1 mb-7"
        cols="12"
      >
        <div
          class="c2 c3"
          color="#EFEFEF"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-tools fa-w-16 fa-3x "
            color="#4A545E"
            data-icon="tools"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M501.1 395.7L384 278.6c-23.1-23.1-57.6-27.6-85.4-13.9L192 158.1V96L64 0 0 64l96 128h62.1l106.6 106.6c-13.6 27.8-9.2 62.3 13.9 85.4l117.1 117.1c14.6 14.6 38.2 14.6 52.7 0l52.7-52.7c14.5-14.6 14.5-38.2 0-52.7zM331.7 225c28.3 0 54.9 11 74.9 31l19.4 19.4c15.8-6.9 30.8-16.5 43.8-29.5 37.1-37.1 49.7-89.3 37.9-136.7-2.2-9-13.5-12.1-20.1-5.5l-74.4 74.4-67.9-11.3L334 98.9l74.4-74.4c6.6-6.6 3.4-17.9-5.7-20.2-47.4-11.7-99.6.9-136.6 37.9-28.5 28.5-41.9 66.1-41.2 103.6l82.1 82.1c8.1-1.9 16.5-2.9 24.7-2.9zm-103.9 82l-56.7-56.7L18.7 402.8c-25 25-25 65.5 0 90.5s65.5 25 90.5 0l123.6-123.6c-7.6-19.9-9.9-41.6-5-62.7zM64 472c-13.2 0-24-10.8-24-24 0-13.3 10.7-24 24-24s24 10.7 24 24c0 13.2-10.7 24-24 24z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
    <h1
      class="c4 mb-4"
    >
      Sorry, there’s been a problem
    </h1>
    <p
      class="c5 mb-7"
    >
      We’re working to fix this problem and we’ll be up and running shortly.
    </p>
    <p
      class="c6 mb-8"
    >
      Keep seeing this page?
       
      <a
        class="c7"
        color="#3B46C4"
        href="http://www.wikihow.com/Clear-Your-Browser%27s-Cookies"
        target="_blank"
      >
        Try clearing your browser cookies
      </a>
       
      or come back later
    </p>
    <div
      class="c0"
      cols="12"
      direction="row"
    >
      <div
        class="c1"
        cols="12"
      >
        <a
          class="c7 c8 mb-10"
          color="#3B46C4"
          href="https://www.zopa.com"
        >
          Go to Zopa home
        </a>
      </div>
    </div>
    <p
      class="c6"
    >
      Error code: 500
    </p>
  </div>
</div>
`;
